packages/@react-spectrum/combobox/docs/ComboBox.mdx

```diff
@@ -564,6 +563,83 @@ function AsyncLoadingExample() {
 }
 ```
 
+When both `inputValue` and `selectedKey` are controlled, and the `selectedKey` is set to an initial value before the `items` load, you must update the `inputValue` when the `items` load.
+This can be done by returning `filterText` from the `useAsyncList` load function.
+
+The example below demonstrates how you could update the input value for your selected key once your list of items has been fetched.
+
+```tsx example
+function AsyncLoadingExample() {
+  let [isFocused, setFocused] = React.useState(false);
+  let list = useAsyncList({
+    async load({signal, cursor, filterText, selectedKeys}) {
+      if (cursor) {
+        cursor = cursor.replace(/^http:\/\//i, 'https://');
+      }
+
+      // If no cursor is available, then we're loading the first page,
+      // filtering the results returned via a query string that
+      // mirrors the ComboBox input text.
+      // Otherwise, the cursor is the next URL to load,
+      // as returned from the previous page.
+      let res = await fetch(cursor || `https://swapi.dev/api/people/?search=${filterText}`, {signal});
+      let json = await res.json();
+
+      let selectedText;
+      let selectedKey = selectedKeys.values().next().value;
+
+      // If selectedKey exists and combobox isn't focused, update the input value with the selected key text
+      // This allows the input value to be up to date when items load for the first time or the selected key text is updated server side.
+      if (!isFocused && selectedKey) {
```

Comment:
Well, it does update on blur in this case since `onSelectionChange` is called on blur in `useComboBoxState` since this is a multiple controlled prop combobox. Since `onSelectionChange` in this example is:
```
  let onSelectionChange = (key) => {
    let itemText = list.getItem(key)?.name;
    list.setSelectedKeys(new Set([key]));
    list.setFilterText(itemText);
  };
```  
the combobox value will be updated to "Luke Skywalker" on blur. I can try changing it to be based on `loadingState`, I guess I'll add it to one of the params `load` accepts in `useAsyncList`

Reply:
Do you think that makes sense? I guess the only case where it wouldn't work would be if the user focused the input before the items loaded but didn't type anything? But maybe that's ok? Just trying to think which is easier to explain/document.